test/src/edu/stanford/nlp/util/ArrayUtilsTest.java

```diff
@@ -92,6 +106,433 @@ public void testgetSubListIndex() {
     String[] t8 = {"cough","aggravated","cough","increased","and","cough", "increased","and","cough","and","increased"};
     assertEquals(2, ArrayUtils.getSubListIndex(t7, t8).get(0).intValue());
     assertEquals(5, ArrayUtils.getSubListIndex(t7, t8).get(1).intValue());
+    String[] t9 = {"cough","aggravated","cough","increased","and","cough", "increased","and","cough","and","increased"};
+    String[] t10 = {"cough","increased"};
+    assertEquals(null, ArrayUtils.getSubListIndex(t9, t10));
   }
 
+  @Test
+  public void testFlatten() {
+    double[][] input = {{1.0}, {2.0}, {3.0, 4.0}};
+    double[] flattened = ArrayUtils.flatten(input);
+    assertEquals(4, flattened.length);
+    assertEquals(3.0, flattened[2], 0.0);
+  }
+
+  @Test
+  public void testTo2D() {
+    double[] input = {1.0, 2.0, 3.0, 4.0};
+    double[][] in2d = ArrayUtils.to2D(input, 2);
+    assertEquals(2, in2d.length);
+  }
+
+  @Test
+  public void testRemoveAtDouble() {
+    double[] input = {1.0, 2.0, 3.0, 4.0};
+    double[] nullInput = null;
+    double[] edited = ArrayUtils.removeAt(input, 3);
+    double[] nothingToRemove = ArrayUtils.removeAt(input, 100);
+    boolean inputWasNotModified = input.length == 4 && input[3] == 4.0;
+    assertTrue("Removal should not modify input array", inputWasNotModified);
+    assertEquals(3, edited.length);
+    assertEquals(4, nothingToRemove.length);
+    assertEquals(null, ArrayUtils.removeAt(nullInput, 5));
+  }
+  
+  @Test
+  public void testToString() {
+    int[][] input = {{2, 3, 5, 7}, {11, 13, 17, 19}};
+    assertEquals("[[2, 3, 5, 7],[11, 13, 17, 19]]", ArrayUtils.toString(input));
+  }
+
+  @Test
+  public void testEqualContent() {
+    int[][] input = {{2, 3, 5, 7}, {11, 13, 17, 19}};
+    int[][] sameInput = {{2, 3, 5, 7}, {11, 13, 17, 19}};
+    int[][] differentContent = {{2, 4, 5, 7}, {11, 13, 17, 19}};
+    int[][] differentLength = {{2, 4, 5, 7}, {11, 13, 17, 19}, {200}};
+    int[][] emptyInput = null;
+    assertFalse(ArrayUtils.equalContents(input, differentContent));
+    assertFalse(ArrayUtils.equalContents(input, differentLength));
+    assertFalse(ArrayUtils.equalContents(input, emptyInput));
+    assertFalse(ArrayUtils.equalContents(emptyInput, input));
+    assertTrue(ArrayUtils.equalContents(input, sameInput));
+    assertTrue(ArrayUtils.equalContents(null, emptyInput));
+  }
+
+  @Test
+  public void testEqualsDouble() {
+    double[][] input = {{2.0, 3.0, 5.0, 7.0}, {11.0, 13.0, 17.0, 19.0}};
+    double[][] sameInput = {{2.0, 3.0, 5.0, 7.0}, {11.0, 13.0, 17.0, 19.0}};
+    double[][] differentContent = {{2.0, 4.0, 5.0, 7.0}, {11.0, 13.0, 17.0, 19.0}};
+    double[][] differentLength = {{2.0, 4.0, 5.0, 7.0}, {11.0, 13.0, 17.0, 19.0}, {200.0}};
+    double[][] emptyInput = null;
+    assertFalse(ArrayUtils.equals(input, differentContent));
```

Comment:
One small suggestion, adding test messages, so we know what fails if it fails. removing a test smell.